pkg/api/pod/util.go

```diff
@@ -23,6 +23,18 @@ import (
 	"k8s.io/kubernetes/pkg/util/intstr"
 )
 
+const (
+	// The annotation value is a string specifying the hostname to be used for the pod e.g 'my-webserver-1'
+	PodHostnameAnnotation = "pod.beta.kubernetes.io/hostname"
```

Comment:
https://github.com/kubernetes/kubernetes/pull/20688/files#diff-bf28da68f62a8df6e99e447c4351122dR1312

You guessed right. hostname defaults to pod name.

Every dns segment is independently validated. The overall length is validated as well.

Like I said in my previous comment (or code link in comment), the fully qualified hostname would be <hostname>.my-web-service.<namespace>.svc.<cluster domain>

If subdomain annotation is not specified, the hostname would be just the hostname annotation, or the pod.Name(default)